avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroSerializer.java

```diff
@@ -50,7 +50,19 @@ public void configure(Map<String, ?> configs, boolean isKey) {
     if (maxSchemaObject == null) {
       schemaRegistry = new CachedSchemaRegistryClient(
           (String) url, AbstractKafkaAvroSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_DEFAULT);
-    } else {
+    } else if (maxSchemaObject instanceof String) {
+      try {
+        Integer i = Integer.parseInt((String)maxSchemaObject);
```

Comment:
This still looks like the manual approach -- I think this should be changed to a `configure(serializerConfig(props))` too?